### PR TITLE
Feature/3105  handle tlm for app collabora

### PR DESCRIPTION
**Pull request by Skylsmoi**
https://github.com/tracim/tracim/issues/3105

## Checkpoints

The goal of this section is to help code reviewers and the merge request author to do the required checks on this pull request. Please don't edit this list.
If one or more of these checkpoints are out of scope, please write below why they are.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, then there are associated issues and issue is referenced in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present
- [x] Automated tests have been written (covering the feature or non regression), or *at least* an issue has been created to implement the test

**For developers**

- [x] Manual tests done to ensure stability on whole application layers, issue expectation follow
- [x] The original issue has been updated with a short summary of the implemented solution

**For the quality team**

- [x] Tested by the quality team

### CODE REVIEW

File: frontend_app_collaborative_document_edition/src/container/PopupCreateOfficeDocument.jsx [hunk 1]
```diff
@@ -41,34 +49,55 @@ class PopupCreateCollaborativeDocument extends React.Component {
...
+
+    if (Number(data.content.parent_id) !== Number(state.folderId) ||
+      state.loggedUser.user_id !== data.author.user_id ||
+      state.newContentName !== data.content.label
+    ) return
+
+    this.handleClose()
+
+    GLOBAL_dispatchEvent({
+      type: CUSTOM_EVENT.REDIRECT,
```
Comment (gbonaspetti) [inline frontend_app_collaborative_document_edition/src/container/PopupCreateOfficeDocument.jsx:85]: Should that be here?

File: frontend_app_collaborative_document_edition/src/container/PopupCreateOfficeDocument.jsx [hunk 2]
```diff
@@ -41,34 +49,55 @@ class PopupCreateCollaborativeDocument extends React.Component {
...
   componentDidMount () {
-    document.addEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
     this.setHeadTitle()
     this.setDocumentOptions()
   }
 
-  componentWillUnmount () {
-    document.removeEventListener(CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, this.customEventReducer)
+  handleAllAppChangeLanguage = data => {
+    console.log('%c<PopupCreateHtmlDocument> Custom event', 'color: #28a745', CUSTOM_EVENT.APP_CUSTOM_EVENT_LISTENER, data)
```
Comment (gbonaspetti) [inline frontend_app_collaborative_document_edition/src/container/PopupCreateOfficeDocument.jsx:68]:
```suggestion
    console.log('%c<PopupCreateHtmlDocument> Custom event', 'color: #28a745', CUSTOM_EVENT.ALL_APP_CHANGE_LANGUAGE, data)
```

File: frontend_app_collaborative_document_edition/test/apiMock.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const nock = require('nock')
+
+const mockGetCollaborativeDocumentTemplates200 = (apiUrl) => {
+  return nock(apiUrl)
+    .get(`/collaborative-document-edition/templates`)
```
Comment (gbonaspetti) [inline frontend_app_collaborative_document_edition/test/apiMock.js:5]:
```suggestion
    .get('/collaborative-document-edition/templates')
```

File: frontend_app_collaborative_document_edition/test/apiMock.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    .reply(200, {
+      file_templates: ["default.odp", "default.ods", "default.odt"]
```
Comment (gbonaspetti) [inline frontend_app_collaborative_document_edition/test/apiMock.js:7]:
```suggestion
      file_templates: ['default.odp', 'default.ods', 'default.odt']
```

File: frontend_lib/test/fixture/tracimLiveMessage/commentTlm.js [hunk 5]
```diff
@@ -1,9 +1,10 @@
 import { author } from '../author.js'
+import { CONTENT_TYPE } from '../../../src'
```
Comment (gbonaspetti) [inline frontend_lib/test/fixture/tracimLiveMessage/commentTlm.js:2]: Is this import right? Maybe it's:
 ```suggestion
import { CONTENT_TYPE } from '../../../src/helper.js'
```

File: frontend_lib/test/fixture/tracimLiveMessage/fileTlm.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { author } from '../author.js'
+import { CONTENT_TYPE } from '../../../src'
```
Comment (gbonaspetti) [inline frontend_lib/test/fixture/tracimLiveMessage/fileTlm.js:2]: Is this import right? Maybe it's:
 ```suggestion
import { CONTENT_TYPE } from '../../../src/helper.js'
```

**Comment by Skylsmoi** [comment]
note: all tests are green. This PR's page doesn't seem to update the status

**Comment by PhilippeAccorsi** [comment]
All seems to work correctly on branch.

**Comment by PhilippeAccorsi** [comment]
@Skylsmoi you can merge after adding your correction about build app (I think you can skip ci)

**Pull request merged**